src/main/kotlin/com/coder/gateway/views/steps/CoderWorkspaceStepView.kt

```diff
@@ -89,7 +89,7 @@ class CoderWorkspaceStepView(
     // Called with a boolean indicating whether IDE selection is complete.
     private val nextButtonEnabled: (Boolean) -> Unit,
 ) : CoderWizardStep {
-    private val cs = CoroutineScope(Dispatchers.Main)
+    private val cs = CoroutineScope(Dispatchers.IO)
```

Comment:
I moved it to `IO` instead of `Main` because the app window seemed to be using `Main` so the IDE lazy-loaded selections were never running//loading.  So I have no idea if this is correct or not...

Reply:
@coryb is this code performing an I/O heavy operation, like waiting on network calls or disk read / writes? If not, `Dispatchers.Unconfined` might be okay. More info here: https://kotlinlang.org/docs/coroutine-context-and-dispatchers.html#dispatchers-and-threads

Reply:
The operation I think uses ssh to connect to the workspace to look for installed ides, so it is certainly using network I/O, but probably does not qualify as "I/O heavy"

Reply:
If that’s the case, the IO Dispatcher is likely safe to keep as-is 👍🏼

Reply:
There probably is some magic to get `Main` working (for example we had a similar issue when launching from `File > Remote Development` and the fix was to add the modality state to `launch`), but this seems good to me!  I had the general impression that updating UI from `IO` was a no-go (and here we update foreground colors, text, etc) but it seems to work fine as far as I can tell.  :shrug: